### ISSUE TITLE
Misunderstanding out of sample prediction 

### ISSUE THREAD

**Issue by firmai** [awslabs/gluon-ts#584]
It's fairly simple what I want to do, I want to train a model, and I want to predict out of sample. Why does predictor.predict() want input data? I can't give it input data, because it is out of sample? Does it just want a simple dataframe, can it be any? It doesn't seem so as the results change depending on the input data. When I add data [1,1,1,1,1,1] it gives completely the wrong predictions.  What am I missing. How can this possibly be this hard? 

gluonts_dataframe -> my own function to convert to your datalist
```
estimator = DeepAREstimator(freq=freq, prediction_length=forecast_len, trainer=Trainer(epochs=20)) #use_feat_dynamic_real=True

models_dict["Gluonts"] = estimator.train(training_data=gluonts_dataframe(train)) 

future = ListDataset([{"target": test[-forecast_len:], "start": test.index[-1] + relativedelta(months=1)}],freq=freq)
#future = ListDataset([{"target": [test[-1]]*forecast_len, "start": test.index[-1] + relativedelta(months=1)}],freq=freq)          
          
for test_entry, forecast in zip(future, model.predict(future)): #next(predictor.predict(future))
    forecast_dict[name] = forecast.samples.mean(axis=0) # .quantile(0.5)
```

**Comment by jgasthaus** [comment]
**TL;DR:** If you want predictions for the same time series as your training data set, starting from where the training set ends, pass the training data set also to ```.predict()```.

> I have used all the other tools, pdarima, prophet etc. with them it made sense, here I don't get it

That may be the reason. :) 

Unlike these other tools, GluonTS focusses on *global* models, i.e. models that are trained on large data sets of related time series, and which learn a mapping from time series' past values (and associated features) to their future. In the GluonTS view, a ```Predictor``` does exactly that: it takes past values of a time series and maps them to a prediction, so that ```predict()``` requires you to pass the past values for the time series you want to make predictions for (these can potentially be different from the ones the model was trained on).

**Comment by firmai** [comment]
Excellent thanks man, I have now come to do that, two details remain somewhat unclear. This is how it looks currently:

1. Should I pass it the entire train dataset at prediction? Or only pass the length of my prediction ```"target": train[-forecast_len:]```
1. For a monthly dataset should I add an additional month to train end ```df.index[-1] + relativedelta(months=1)}]```
1. Does all your and self implemented models follow this prediction protocol?
1. If you can pass it new data at the prediction step does it train twice? Would it not be better to give it two distinct datasets to avoid overfitting?
1. I am using "M" as a frequency as "MS" is disallowed, would this have any negative effects? 
1. Is there a formal way to ensure that the predictor is using the best model, a way for the training to stop at the optimal epoch? Or should I just loop and see for myself. 

```python
estimator = DeepAREstimator(freq=freq, prediction_length=forecast_len, trainer=Trainer(epochs=20))
estimator.train(training_data=gluonts_dataframe(train))
future = ListDataset([{"target": train[-forecast_len:], "start": df.index[-1] + relativedelta(months=1)}],freq=freq)
for df_entry, forecast in zip(future, model.predict(future)): 
   forecast_dict[name] = forecast.samples.mean(axis=0)
```

Thank you that helps a lot; just to squeeze you for a last bit of information, is there a primer somewhere on the web to learn more about the difference of global and I guess local TS models?

**Comment by greenspray** [comment]
That makes sense but what if there is no data between the training data and inference data? I am using some dynamic features that are dependent on the target and if the target was not observed in that gap then the dynamic features are also Null which the model doesn't allow.

**Comment by timoschowski** [comment]
We wrote some high-level introduction for deep learning for time series in Foresight:  
https://ideas.repec.org/a/for/ijafaa/y2018i50p35-41.html
https://ideas.repec.org/a/for/ijafaa/y2018i51p42-47.html

The second one explains what a global model is on high-level terms.

**Comment by greenspray** [comment]
@timoschowski your comment does help much. Perhaps you could share your finding more directly? Furthermore, the articles you linked are not openly accessible :(
